src/Host/Client/Impl/Transports/WebSocketClient.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
```

Comment:
This entire file is from:

https://github.com/aspnet/WebSockets/tree/8b25152a67c64d03e86ed2b2347650c78b441529/src/Microsoft.AspNetCore.WebSockets.Client

They don't offer a NuGet package, but since it's a self-contained class, there are no modifications necessary to just embed it directly into the project.